publication-rest/src/main/java/no/unit/nva/publication/create/CreatePublicationHandler.java

```diff
@@ -99,21 +104,33 @@ private UserInstance createUserInstanceForInternalUser(RequestInfo requestInfo)
 
     private UserInstance createUserInstanceFromLoginInformation(RequestInfo requestInfo) throws UnauthorizedException {
         return requestInfo.clientIsThirdParty()
-                   ? RequestUtil.createExternalUserInstance(requestInfo, identityServiceClient)
-                   : createUserInstanceForInternalUser(requestInfo);
+            ? RequestUtil.createExternalUserInstance(requestInfo, identityServiceClient)
+            : createUserInstanceForInternalUser(requestInfo);
     }
-    
+
     private ResourceOwner createInternalResourceOwner(RequestInfo requestInfo) throws UnauthorizedException {
         return attempt(() -> requestInfo.getTopLevelOrgCristinId().orElseThrow())
-                   .map(topLevelOrgCristinId -> new ResourceOwner(new Username(requestInfo.getUserName()),
-                                                                               topLevelOrgCristinId))
-                   .orElseThrow(fail -> new UnauthorizedException());
+            .map(topLevelOrgCristinId -> new ResourceOwner(new Username(requestInfo.getUserName()),
+                topLevelOrgCristinId))
+            .orElseThrow(fail -> new UnauthorizedException());
     }
-    
+
     private void setLocationHeader(SortableIdentifier identifier) {
         addAdditionalHeaders(() -> Map.of(
             HttpHeaders.LOCATION,
             getLocation(identifier).toString())
         );
     }
+
+    private boolean isThesisAndHasNotPublishThesis(RequestInfo requestInfo, CreatePublicationRequest request) {
+        var thesisKinds = List.of("DegreeBachelor", "DegreeMaster", "DegreePhd");
+        var kind =
+            attempt(() -> request.getEntityDescription().getReference().getPublicationInstance().getInstanceType())
+                .toOptional();
+
+        if (kind.isPresent() && thesisKinds.contains(kind.get())) {
+            return !requestInfo.userIsAuthorized(PUBLISH_THESIS.name());
+        }
+        return false;
```

Comment:
Maybe rename kind to publicationInstanceType and thesisKinds to static list THESES?


        if (isThesis(publicationInstanceType)) {
            hasRightsToPublicateThesis(requestInfo);
        }